source/Nevermore/Advanced/ReadTransaction.cs

```diff
@@ -586,6 +644,8 @@ void AddCommandTrace(string commandText)
                 if (commandTrace.Count <= 200)
                     commandTrace.Add(DateTime.Now.ToString("s") + " " + commandText);
             }
+
+            customCommandTrace?.Invoke(commandText);
```

Comment:
When this is provided, we _could_ disable the default command tracing behaviour and leave it up to the consumer only. However, other things that are tied to this still exist like `RelationalTransactionRegistry`, and there's no guarantee the consumer doesn't still want that stuff. So, I think it's safest to just leave it be.

Reply:
An alternative could be to make it the default implementation. That might get tricky to get parity with the current locking behaviour though. It looks like we hold a collection of traces for the lifetime of this transaction and also use it as a locking mechanism?

Reply:
> It looks like we hold a collection of traces for the lifetime of this transaction and also use it as a locking mechanism?

Yep 👍 and Nevermore relies on that collection elsewhere like in `RelationalTransactionRegistry`, so turning it off when we provide a custom implementation would stop that from working. It may be solvable but I think it's best to keep it an additional thing right now.